src/general.py

```diff
@@ -36,18 +37,114 @@ def get_resources() -> dict:
     return unique_resources
 
 
-def run_exerciser(tests_dir: Path, working_dir: Path, run=False):
+def run_exerciser(args: argparse.Namespace):
     """
-    Usage: calls necessary methods for setting up, running, and cleaning up exerciser
-    @param tests_dir: directory containing all exerciser tests
-    @param working_dir: directory for storing info on exerciser runs
-    @param run: whether or not to create working file system and run tests
+    Usage: interprets cla and runs the exerciser based on the provided options
+    @param args: program arguments as returned by parse_cla() in __main__
+    """
+    # -w option
+    # changes location of working_dir, exiting if the dir dne
+    if args.change_working_dir == None:
+        working_dir = Path("working")
+    else:
+        working_dir = Path(args.change_working_dir)
+        if not os.path.exists(working_dir):
+            print("Error: Working dir must exist. Check arg val for -w and try again")
+            print("Exiting...")
+            sys.exit(1)
+
+    # -t option
+    # changes location of tests_dir, exiting if the dir dne
+    if args.change_test_dir == None:
+        tests_dir = Path("tests")
+    else:
+        tests_dir = Path(args.change_test_dir)
+        if not os.path.exists(tests_dir):
+            print("Error: Test dir must exist. Check arg val for -t and try again")
+            print("Exiting...")
+            sys.exit(1)
+
+    # -s option
+    # prints the list of currenlt available resources to the command line
+    if args.snapshot:
+        print("Here is a list of all currently available GLIDEIN_ResourceName s:")
+        resources = get_resources()
+        for resource in resources.keys():
+            print(resource)
+        print("End of resource list")
+
+    # -p option
+    # prints all the test names in tests_dir to the command line
+    if args.print_tests:
+        print("Here is a list of all tests in the test dir:")
+        for test in tests_dir.iterdir():
+            print(test.name)
+        print("End of test list")
+
+    # -f option
+    # clears the working_dir
+    if args.flush_all:
+        print("Flushing entire working directory")
+        for item in working_dir.iterdir():
+            shutil.rmtree(item)
+
+    # -d option
+    # clears all timestamp_dirs in working_dir older than the date provided
+    # will exit with an err if the date is not provided in the requested format
+    if (not args.flush_by_date == None) and (not args.flush_all):
```

Comment:
This should be an `if/elif` statement because having a following `if` statement that says `(not args.flush_all)` is having the code do the same check twice while making readability more difficult. Plus, we may add more flush variations in the future (default execution flush date) 
Also, do `args.flush_by_date is not None` rather than `(not args.flush_by_date == None)`